core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateProviderImpl.java

```diff
@@ -260,12 +255,13 @@ public void run() {
                }
                try {
                   Flowable.fromPublisher(stProvider.entryPublisher(k -> shouldSendKey(k) && !dataContainer.containsKey(k), true, true))
-                        .map(XSiteState::fromCacheLoader)
-                        .takeUntil(l -> canceled)
-                        .buffer(chunkSize <= 0 ? Integer.MAX_VALUE : chunkSize)
-                        // We want the CacheException to be thrown to the catch block
-                        .blockingForEach(l -> {
+                          .map(XSiteState::fromCacheLoader)
+                          .takeUntil(l -> canceled)
+                          .buffer(chunkSize)
+                          // We want the CacheException to be thrown to the catch block
+                          .blockingForEach(l -> {
                            try {
+                              // TODO Make non-blocking and use concatMapCompletable like OutboundTransferTask
```

Comment:
Is this going to be covered in a different PR or were you planning on in this one?

Reply:
Different Pr